app/api/cron/daily-sales-summary/process-daily-sales-summary.ts

```diff
@@ -0,0 +1,263 @@
+import {
+	agentActivities,
+	agents,
+	db,
+	subscriptions,
+	teams,
+	userSeatUsageReports,
+} from "@/drizzle";
+import { stripe } from "@/services/external/stripe";
+import * as csv from "@fast-csv/format";
+import { captureException } from "@sentry/nextjs";
+import { and, eq, gte, lt } from "drizzle-orm";
+import type Stripe from "stripe";
+import invariant from "tiny-invariant";
+import {
+	formatAgentTimeUsageEvicence,
+	formatSubscriptionEvidence,
+	formatUserSeatEvidence,
+} from "./formatter";
+import { putCSV } from "./put-csv";
+import type {
+	SubscriptionCreateInvoiceEvidence,
+	SubscriptionCycleInvoiceEvidence,
+} from "./types";
+
+/**
+ * Process daily sales summary for the target date
+ *
+ * @param targetDate - Date object set to 00:00:00 UTC
+ */
+export async function processDailySalesSummary(targetDate: Date) {
+	// fetch invoices created on the target date
+	const periodStart = targetDate.getTime() / 1000;
+	const periodEnd = targetDate.getTime() / 1000 + 24 * 60 * 60;
```

Comment:
📝 For My Notes

I have confirmed that the unit of `stripe.invoices.created` is in seconds.

https://docs.stripe.com/api/invoices/list?lang=node

```json
      "created": 1680644467,
```

<img width="2032" alt="image" src="https://github.com/user-attachments/assets/7c452c35-b851-42fd-851c-890e5b2dd5b2" />